xarray/core/accessor_str.py

```diff
@@ -574,15 +706,31 @@ def match(self, pat, case=True, flags=0):
         -------
         matched : array of bool
         """
-        if not case:
-            flags |= re.IGNORECASE
+        is_compiled_re = self._check_is_compiled_re(pat)
```

Comment:
Isn't this code block equal to `_re_compile`?

Reply:
Yes, I missed that one.  Fixed.